cmd/kubeadm/app/util/pkiutil/pki_helpers.go

```diff
@@ -142,6 +158,26 @@ func WriteCert(pkiPath, name string, cert *x509.Certificate) error {
 	return nil
 }
 
+// WriteCertBundle stores the given certificate bundle at the given location
+func WriteCertBundle(pkiPath, name string, certs []*x509.Certificate) error {
+	for _, cert := range certs {
+		if cert == nil {
+			return errors.New("certificate cannot be nil when writing to file")
```

Comment:
can we include an index `i` in this loop and modify the error to the following:

```
return errors.Errorf("found nil certificate at position %d when writing bundle to file", i)
```

Reply:
Done in this commit: https://github.com/kubernetes/kubernetes/pull/97266/commits/945ca604f4d26f8805cc390191b29b7da5db761d